modules/modals/utils.ts

```diff
@@ -0,0 +1,78 @@
+interface ModalConfig {
+  modalId: string;
+  openButtonId?: string;
+  closeButtonId: string;
+}
+
+export const useModal = ({
+  modalId,
+  openButtonId,
+  closeButtonId,
+}: ModalConfig) => {
+  const modal = document.getElementById(modalId);
+  const openModalButton = openButtonId
+    ? document.getElementById(openButtonId)
+    : null;
+  const closeModalButton = document.getElementById(closeButtonId);
+  const focusableElements =
+    'button, [href], input, select, textarea, [tabindex]:not([tabindex="-1"])';
+  const focusableContent = modal?.querySelectorAll(focusableElements);
+  const firstFocusableElement = focusableContent?.[0] as HTMLElement;
+  const lastFocusableElement = focusableContent?.[
+    focusableContent.length - 1
+  ] as HTMLElement;
+
+  if (!modal || !closeModalButton) {
+    console.warn('Modal or modal buttons not found');
+    return;
+  }
+
+  const trapFocus = (event: KeyboardEvent) => {
+    const isTabPressed = event.key === 'Tab' || event.keyCode === 9;
+    if (!isTabPressed) return;
+
+    if (event.shiftKey && document.activeElement === firstFocusableElement) {
+      lastFocusableElement?.focus();
+      event.preventDefault();
+    } else if (
+      !event.shiftKey &&
+      document.activeElement === lastFocusableElement
+    ) {
+      firstFocusableElement?.focus();
+      event.preventDefault();
+    }
+  };
+
+  const setInert = (state: any) => {
+    document.querySelectorAll('body > *:not(.modal)').forEach((element) => {
+      element.classList.toggle('inert', state);
+    });
+  };
+
+  const showModal = (customShowModal?: () => void) => {
+    if (!modal) return;
+    modal.classList.add('open');
+    modal.style.display = 'block';
+    modal.setAttribute('aria-hidden', 'false');
+    setInert(true);
+    modal.addEventListener('keydown', trapFocus);
+    firstFocusableElement?.focus();
+
+    customShowModal && customShowModal();
```

Comment:
This could be implemented in a better way...

Reply:
I ended up ripping this out entirely. It didn't really make sense, and it wasn't reasonably clear to the user of `useModal` that passing more to show/hide was even an option. All around it was a clunky solution to a singular problem - whenever a modal closes, and there is a Form element in that modal, the contents of the form should be reset. 

This wasn't something we needed anything overly complex for - a modal can or cannot have a form in it, so I added an optional `modalFormId` prop to `useModal`. That way, if there happens to be a form in the modal we can specify its `id` and reset whenever hideModal is called. 

Of course there are plenty of "what ifs" with this approach - what if we need to do more than reset a form? What if there's more than one form? - all valid, but a non-issue in the context of this project at this moment. We'll cross that bridge if we ever come to it, and I am of the opinion adding this prop is reasonable, and fairly straightforward to read/understand